pkg/common/configs/configvalidator.go

```diff
@@ -673,6 +677,22 @@ func checkQueues(queue *QueueConfig, level int) error {
 	return nil
 }
 
+func IsQueueNameValid(queueName string) error {
+	if !QueueNameRegExp.MatchString(queueName) {
+		return fmt.Errorf("invalid child name '%s', a name must only have alphanumeric characters,"+
+			" - or _, and be no longer than 64 characters", queueName)
+	}
+	return nil
+}
+
+func IsQueuePathValid(queuePath string) error {
```

Comment:
Not sure how this would work as we allow dots which means the length cannot be limited as each part can be 64 characters and I could have 10, 20 or more levels in a queue.

These valid queue fails this check:
`root.parent_lvl1.parent_lvl2.parent_lvl3.parent_lvl4.parent_lvl5.parent_lvl6.parent_lvl7.parent_lvl8.parent_lvl9.parent_lvl10.leaf`
or
`root.company_name.a-long-namespace-name.wilfred_dot_spiegelenburg`

Checking each part before building the path should be enough to handle this. See further comments below

Reply:
Yes, length check is not required. However, I still see a need to do validate that code path because there is an option to pass the full queue path in the config. If we really want to avoid full queuePath check, we should split the string based on dot and apply queueName check itself for each part.